Sources/SwifterSwift/UIKit/UIViewExtensions.swift

```diff
@@ -49,6 +49,14 @@ public extension UIView {
         /// SwifterSwift: easeInOut animation.
         case easeInOut
     }
+
+    /// SwifterSwift: Apply gradient directions
+    enum GradientDirection {
```

Comment:
I'm wondering if this should be a struct, e.g.
```swift
struct GradientDirection {
    static let topToBottom = GradientDirection(startPoint: CGPoint(x: 0.5, y: 0.0),
                                               endPoint: CGPoint(x: 0.5, y: 1.0))
    static let bottomToTop = GradientDirection(startPoint: CGPoint(x: 0.5, y: 1.0),
                                               endPoint: CGPoint(x: 0.5, y: 0.0))
    static let leftToRight = GradientDirection(startPoint: CGPoint(x: 0.0, y: 0.5),
                                               endPoint: CGPoint(x: 1.0, y: 0.5))
    static let rightToLeft = GradientDirection(startPoint: CGPoint(x: 1.0, y: 0.5),
                                               endPoint: CGPoint(x: 0.0, y: 0.5))

    let startPoint: CGPoint
    let endPoint: CGPoint
}
```
This way it can be customised to exact values.

Reply:
I added a computed property in `GradientDirection` enum to make it more 'exact', so there is no need to add an extra struct to achieve this.

```swift
enum GradientDirection {
    case topToBottom
    case bottomToTop
    case leftToRight
    case rightToLeft

    var gradientEndpoint: (CGPoint, CGPoint) {
        switch self {
        case .topToBottom:
            return (CGPoint(x: 0.5, y: 0.0), CGPoint(x: 0.5, y: 1.0))
        case .bottomToTop:
            return (CGPoint(x: 0.5, y: 1.0), CGPoint(x: 0.5, y: 0.0))
        case .leftToRight:
            return (CGPoint(x: 0.0, y: 0.5), CGPoint(x: 1.0, y: 0.5))
        case .rightToLeft:
            return (CGPoint(x: 1.0, y: 0.5), CGPoint(x: 0.0, y: 0.5))
        }
    }
}
```

Reply:
But what if somebody wants to customise the direction? This is limiting.
Also, you've used a tuple instead of the struct, which is harder to read when using and is doing the same thing.